Update docker_remote_api_v1.22.md

ZJUshuaizhou opened the pull request:
related issue #16849 
@thaJeztah 
I am so sorry for that,it is my first time to do like this! I was work so hard,i don't know why the git in my computer does not work well !Excuse me! I can do it well this time !:)

The ‘Example response’ at the ‘Get container stats based on resource usage’ section.Update the response with the new one which contains the precpu_stats field and give some explains

Signed-off-by: ZJUshuaizhou 21551191@zju.edu.cn

ZJUshuaizhou commented:
@coolljt0725 
#20490 
#20488 
#20485

coolljt0725 commented:
@ZJUshuaizhou  Please follow the comment https://github.com/docker/docker/pull/20490#issuecomment-186202882 by @thaJeztah

ZJUshuaizhou commented:
I test the api use the computer with the 4-core cpu,and update the docs v1.22,can you check it?
@thaJeztah

ZJUshuaizhou commented:
@vdemeester

vdemeester commented on docs/reference/api/docker_remote_api_v1.22.md:
```diff
@@ -867,19 +867,35 @@ This endpoint returns a live stream of a container's resource usage statistics.
+            "throttling_data" : {"periods":0,"throttled_periods":0,"throttled_time":0}
          }
       }
+The precpu_stats is the cpu statistic of last read, which is used for calculating the cpu usage percent. It is not the exact copy of the “cpu_stats”field.
```
I think you need to add a line between the JSON and this sentence (for the formatting to be ok)

ZJUshuaizhou replied:
yeah！iupdate

ZJUshuaizhou replied:
is it right now?

vdemeester replied:
yes ;)

ZJUshuaizhou replied:
yeah！so，can you merge it？i hava communicated with @coolljt0725 today，he said it is right ! thanks! :)

thaJeztah commented:
> so，can you merge it

Were you planning to also do the other versions of the API documentation?

(this change LGTM)

ZJUshuaizhou commented:
yes！My plan is that if you agree with the changes in the doc, i will change the other versions of API documentation like  this ! :)

GordonTheTurtle commented:
Job: Docker-PRs-Win2Lin [FAILED](https://jenkins.dockerproject.org/job/Docker-PRs-Win2Lin/22062/console):

~~~console

---
ttachSilent	1.711s
PASS: docker_cli_start_test.go:112: DockerSuite.TestStartMultipleContainers	2.178s
PASS: docker_cli_start_test.go:96: DockerSuite.TestStartPausedContainer	0.665s
PASS: docker_cli_start_test.go:71: DockerSuite.TestStartRecordError	2.216s
PASS: docker_api_containers_test.go:1433: DockerSuite.TestStartWithNilDNS	0.463s
PASS: docker_api_containers_test.go:890: DockerSuite.TestStartWithTooLowMemoryLimit	0.326s

----------------------------------------------------------------------
FAIL: docker_cli_stats_test.go:98: DockerSuite.TestStatsAllNewContainersAdded

docker_cli_stats_test.go:131:
    c.Fatal("failed to observe new container created added to stats")
... Error: failed to observe new container created added to stats


----------------------------------------------------------------------
PASS: docker_cli_stats_test.go:80: DockerSuite.TestStatsAllNoStream	3.504s
PASS: docker_cli_stats_test.go:56: DockerSuite.TestStatsAllRunningNoStream	3.978s
PASS: docker_cli_stats_test.go:43: DockerSuite.TestStatsContainerNotFound	3.119s
PASS: docker_cli_stats
---

~~~

GordonTheTurtle commented:
Job: Docker-PRs-experimental [FAILED](https://jenkins.dockerproject.org/job/Docker-PRs-experimental/15247/console):

~~~console

---
Suite.TestDaemonLoggingDriverDefaultOverride	1.377s
PASS: docker_cli_daemon_test.go:1217: DockerDaemonSuite.TestDaemonLoggingDriverNone	2.031s
/go/src/github.com/docker/docker/hack/make/test-integration-cli: line 18: 11864 Terminated              ( bundle .integration-daemon-start; bundle .integration-daemon-setup; bundle_test_integration_cli; bundle .integration-daemon-stop ) 2>&1
     11865                       | tee -a "$DEST/test.log"
Build step 'Execute shell' marked build as failure
[PostBuildScript] - Execution post build scripts.
[Docker-PRs-experimental] $ /bin/sh -xe /tmp/hudson7674839825674273924.sh
+ set +x


---

~~~

ZJUshuaizhou committed:
update all the versions from v1.19 to v1.23
update v1.22

Signed-off-by: ZJUshuaizhou <21551191@zju.edu.cn>

update all the versions docs

Signed-off-by: ZJUshuaizhou <21551191@zju.edu.cn>

revert v1,20

Signed-off-by: ZJUshuaizhou <21551191@zju.edu.cn>

update v1,20

Signed-off-by: ZJUshuaizhou <21551191@zju.edu.cn>

revert v1,20

Signed-off-by: ZJUshuaizhou <21551191@zju.edu.cn>

update v1,20

Signed-off-by: ZJUshuaizhou <21551191@zju.edu.cn>

ZJUshuaizhou commented:
I have changed the docs all the versions from v1.19 to v1.23 ! the code is changed from docker v1.7 - API doc v1.19 . is the issure completed?Do you have some other suggestions? 
@thaJeztah 
@vdemeester 
thanks :)

thaJeztah commented on docs/reference/api/docker_remote_api_v1.20.md:
```diff
@@ -695,20 +695,37 @@ This endpoint returns a live stream of a container's resource usage statistics.
+            "system_cpu_usage" : 739306590000000,
+            "throttling_data" : {"periods":0,"throttled_periods":0,"throttled_time":0}
+         }
+		 "precpu_stats" : {
```
Looks like the indentation here went wrong

thaJeztah commented on docs/reference/api/docker_remote_api_v1.19.md:
```diff
@@ -682,20 +682,37 @@ This endpoint returns a live stream of a container's resource usage statistics.
-            "throttling_data" : {}
+            "system_cpu_usage" : 739306590000000,
+            "throttling_data" : {"periods":0,"throttled_periods":0,"throttled_time":0}
+         }
```
Oh! Overlooked this the first time; there's a comma missing after `}` here

(Also in the other documents)

thaJeztah commented:
Thanks @ZJUshuaizhou; found two issues (but the comma needs to be changed in all the files)

ZJUshuaizhou commented:
Thanks @thaJeztah you are right ! I fixed it ! is it ok now? : ) @vdemeester

thaJeztah commented:
LGTM

ping @vdemeester :frog:

vdemeester commented:
LGTM 🐯
